src/core/src/package_managers/UbuntuProClient.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2023 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Requires Python 3.5+
+
+"""This is the Ubuntu Pro Client implementation"""
+
+
+class UbuntuProClient:
+    def __init__(self, env_layer, composite_logger):
+        self.env_layer = env_layer
+        self.composite_logger = composite_logger
+        self.ubuntu_pro_client_install_cmd = 'sudo apt-get install ubuntu-advantage-tools -y'
+
+    def install_or_update_pro(self):
+        """install/update pro(ubuntu-advantage-tools) to the latest version"""
+        try:
+            # Install Ubuntu Pro Client.
+            code, output = self.env_layer.run_command_output(self.ubuntu_pro_client_install_cmd, False, False)
+            if code == 0:
+                self.composite_logger.log_debug("Ubuntu Pro Client installation successful. " + output)
+            else:
+                self.composite_logger.log_debug("Ubuntu Pro Client installation failed. " + output)
+        except Exception as error:
+            self.composite_logger.log_debug("Exception in installing Ubuntu Pro Client installation " + repr(error))
+
+    def is_pro_working(self):
+        """check if pro version api returns the version without any errors/warnings."""
+        try:
+            self.composite_logger.log_debug("Ubuntu Pro Client try importing.")
+            from uaclient.api.u.pro.version.v1 import version
+            version_result = version()
+            self.composite_logger.log_debug("Ubuntu Pro Client version: " + version_result.installed_version)
+            return True
+
+        except Exception as error:
+            self.composite_logger.log_debug("Exception in querying pro version api " + repr(error))
+            return False
+
+    def get_security_updates(self):
+        pass
+
+    def get_all_updates(self):
+        pass
+
+    def get_other_updates(self):
+        pass
+
+    def is_reboot_pending(self):
+        """query pro api to get the reboot status"""
+        try:
+            from uaclient.api.u.pro.security.status.reboot_required.v1 import reboot_required
```

Comment:
Has this code been run in the compiled py file (especially on an Ubuntu machine without this import available)?

Reply:
By the time it reaches here, pro-client will be installed. If for some reason, we don`t have the imports at this point, exception ImportError will be thrown, which is handled in the same method. I have verified this with a script locally.

Reply:
To test this scenario specifically, i installed pro client in a VM, but deleted the module files manually (so import will not be possible). When i ran patch operations, i got this below error 
`        DEBUG: Exception in Ubuntu Pro Client api reboot status: ModuleNotFoundError("No module named 'uaclient.api.u.pro.security.status.reboot_required'",)
        DEBUG: is_pro_api_success: False, reboot_status False`

Reply:
Sound good. To sanity-check how the imports land for my peace of mind, do you mind e-mailing me the MsftLinuxPatchCore.py file compiled from the code in the current iteration. I just want to be sure the Packager is working correctly here.

Reply:
Sure. Have sent you email with the output.

Reply:
Looks good. Closing comments.